src/tabpfn/model/loading.py

```diff
@@ -718,3 +738,126 @@ def save_tabpfn_model(model: nn.Module, save_path: Path | str) -> None:
 
     # Save the checkpoint
     torch.save(checkpoint, save_path)
+
+
+def save_fitted_tabpfn_model(estimator: BaseEstimator, path: Path | str) -> None:
+    """Persist a fitted TabPFN estimator to ``path``.
+
+    This stores the initialization parameters and the fitted state, but crucially
+    omits the large foundation model weights for efficiency.
+    """
+    if not hasattr(estimator, "executor_"):
+        raise RuntimeError("Estimator must be fitted before saving.")
+
+    path = Path(path)
+    if path.suffix != ".tabpfn_fit":
```

Comment:
this seems a bit strict no?

Reply:
ah I see in the rest of the code that it's actually needed. I still think it's a bit harsh and we could handle any filename but maybe that's fine